stdlib/string.gr

```diff
@@ -2338,3 +2338,42 @@ provide let toAsciiUppercase = string => {
   }
   implode(chars)
 }
+
+/**
+ * Produces a new string by repeating a given substring a given number of times.
+ *
+ * @param string: The string to repeat
+ * @param count: The number of times to repeat the string
+ * @returns A string containing the repeated input string
+ *
+ * @throws InvalidArgument(String): When the `count` is not an integer
+ * @throws InvalidArgument(String): When the `count` is negative
+ *
+ * @example assert String.repeat("=", 5) == "====="
+ * @example assert String.repeat(".", 0) == ""
+ *
+ * @since v0.6.7
+ */
+@unsafe
+provide let repeat = (string, count) => {
+  use WasmI32.{ (+), (*), (<), (&), (!=) }
+  if ((WasmI32.fromGrain(count) & 1n) != 1n) {
+    throw InvalidArgument("Invalid count value")
+  }
+  let mut rawCount = untagSimpleNumber(count)
+  if (WasmI32.eqz(rawCount)) return ""
+  if (rawCount < 0n) {
+    throw InvalidArgument("Invalid count must be a positive integer or zero")
+  }
+  let mut stringPtr = WasmI32.fromGrain(string)
+  let byteLength = WasmI32.load(stringPtr, 4n)
+  if (WasmI32.eqz(byteLength)) return ""
+  stringPtr += 8n
+  let strPtr = allocateString(byteLength * rawCount)
```

Comment:
We should rename this to something more clear, this is too similar to `stringPtr`